### PR TITLE
K8S 1.22 Updates

### PR DESCRIPTION
**Pull request by DanStough**
Updates for 1.22 and general housekeeping:
* Updated depedencies for kube 1.22
* Moved Rook to latest package that still has support for v1 API
* Move etcd package to the new, maintained repos
* Changes to kubeadm - now using pods to get the remaining ETCD members.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,133 +4,140 @@ go 1.17
...
 	go.undefinedlabs.com/scopeagent v0.1.12
-	k8s.io/api v0.22.3
-	k8s.io/apimachinery v0.22.3
+	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/api v0.22.5
+	k8s.io/apimachinery v0.22.5
 	k8s.io/client-go v12.0.0+incompatible
-	k8s.io/kubernetes v1.17.2
+	k8s.io/kubernetes v1.22.5
+	k8s.io/kubernetes/v121 v121.0.0-00010101000000-000000000000
```
Comment (DanStough) [inline go.mod:23]: Not sure if this is the best practice, but I still need the previous structs to maintain backward compatibility.

File: pkg/cluster/certs.go [hunk 2]
```diff
@@ -14,12 +14,12 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 )
 
-var podFailedErr error = errors.New("pod failed")
+var errPodFailed error = errors.New("pod failed")
```
Comment (DanStough) [inline pkg/cluster/certs.go:17]: go vet complained about the name

File: pkg/cluster/certs.go [hunk 3]
```diff
@@ -131,7 +131,7 @@ func (c *Controller) getRotateCertsPodConfig(nodeName string) *corev1.Pod {
 				},
 			},
 			Containers: []corev1.Container{
-				corev1.Container{
```
Comment (DanStough) [inline pkg/cluster/certs.go:134]: linter complaints

File: pkg/cluster/registry_cert.go [hunk 4]
```diff
@@ -85,14 +87,14 @@ func (c *Controller) readRegistryCert(namespace, name string) (*x509.Certificate
 		return nil, errors.Wrapf(err, "parse registry.crt")
 	}
 	if len(certs) != 1 {
-		return nil, fmt.Errorf("Expected exactly 1 cert in registry.crt, got %d", len(certs))
+		return nil, fmt.Errorf("expected exactly 1 cert in registry.crt, got %d", len(certs))
```
Comment (DanStough) [inline pkg/cluster/registry_cert.go:90]: linting complains about the capitalization

File: pkg/internallb/haproxy_test.go [hunk 5]
```diff
@@ -74,7 +74,7 @@ metadata:
   namespace: kube-system
 spec:
   containers:
-  - image: haproxy:2.4.2
+  - image: haproxy:2.4.10
```
Comment (DanStough) [inline pkg/internallb/haproxy_test.go:77]: Updated for v0.14.0 - should be changed in kurl manifest

**Review by sgalsaleh — APPROVED** [review]

**Pull request merged**